akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/EventPusherConsumerServiceImpl.scala

```diff
@@ -88,9 +90,20 @@ private[akka] object EventPusherConsumerServiceImpl {
                   if (fillSequenceNumberGaps)
                     throw new IllegalArgumentException(
                       s"fillSequenceNumberGaps can not be true for RES (pid ${envelope.persistenceId}, seq nr ${envelope.sequenceNr} from ${replicationId}")
-                  val entityRef = sharding
-                    .entityRefFor(replicationSettings.entityTypeKey, destinationReplicaId.entityId)
-                    .asInstanceOf[EntityRef[PublishedEvent]]
+
+                  val entityRef =
+                    if (replicationSettings.configureEntityWithoutEnvelope.isEmpty) {
+                      sharding
+                        .entityRefFor(replicationSettings.entityTypeKey, destinationReplicaId.entityId)
+                        .asInstanceOf[EntityRef[PublishedEvent]]
+                    } else {
+                      new NoEnvelopeEntityRef[PublishedEvent](
+                        replicationSettings.entityTypeKey.asInstanceOf[EntityTypeKey[PublishedEvent]],
+                        dataCenter = None, // FIXME we don't have access to the Entity here
```

Comment:
Might be possible, but it's very difficult to pass around the Entity (which is used in sharding.init)